compiler/rustc_mir/src/borrow_check/diagnostics/conflict_errors.rs

```diff
@@ -1357,6 +1357,18 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
             ),
         };
         err.span_note(constraint_span, &msg);
+        if let ConstraintCategory::CallArgument = category {
+            if let Some(generator_kind) = use_span.generator_kind() {
+                if let GeneratorKind::Async(_) = generator_kind {
+                    err.note(
+                        "borrows cannot be held across a yield point, because the stack \
+                        space of the current function is not preserved",
+                    );
+                    err.help("see https://rust-lang.github.io/async-book/03_async_await/01_chapter.html#awaiting-on-a-multithreaded-executor \
```

Comment:
I'm worried this link can go stale. It seems like using the error index would be better, we can update this link there if it ever changes.

Reply:
Is [this page](https://doc.rust-lang.org/error-index.html?) the error index? If so, which error are you referring to?

Reply:
That is the error index. I believe @tmandry's idea is to modify https://github.com/rust-lang/rust/blob/9e5f7d5631b8f4009ac1c693e585d4b7108d4275/compiler/rustc_error_codes/src/error_codes/E0373.md to include information from and/or a link to chapter 1 of the async book.

Reply:
I've moved the link to error index in my latest commit.